airbyte-commons/src/test/java/io/airbyte/commons/util/AutoCloseableIteratorsTest.java

```diff
@@ -38,15 +38,18 @@ void testFromIterator() throws Exception {
 
   @Test
   void testFromStream() throws Exception {
-    final Stream<String> stream = spy(Stream.of("a", "b", "c"));
+    final AtomicBoolean isClosed = new AtomicBoolean(false);
```

Comment:
We can't `spy` a `Stream` when `iterator()` is called because it is a `public final` method in Java 17.

This checks the same thing, but in a more future-compatible way.